### ISSUE TITLE
Incorrect parsing of CORE types

### ISSUE THREAD

**Issue by zachriggle** [eliben/pyelftools#93]
CORE files use notes to store register information.  In particular, NT_GNU_ABI_TAG and NT_PRSTATUS are the same value.

Currently, `pyelftools` assumes the former and discards the [data stream](https://github.com/eliben/pyelftools/blob/d7aec6aba2cfed93e69dfe177cc161ccab42f15c/elftools/elf/segments.py#L127).

This should not be done when `elftype` is `"CORE"`.

**Comment by zachriggle** [comment]
As an example, here's one that I finagled so as not to go through the GNU ABI tag parsing;

```
Container({'n_desc': '\x0b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc9\xbc\x00\x00u\xbc\x00\x00\xc9\xbc\x00\x00\xc9\xbc\x00\x00\x00\x00\x00\x00@\x9c\x00\x00\x00\x00\x00\x00\xe0.\x00\x00\x00\x00\x00\x00%\x05\x00\x00\x00\x00\x00\x00\x9c&\x00\x00\xf4\x7frU*`WU\xb8\x98rU\x00\x00\x00\x00\x00\x00\x00\x00\xf4\x7frU\x00\x00\x00\x00+\x00\x00\x00+\x00\x00\x00\x00\x00\x00\x00c\x00\x00\x00\xff\xff\xff\xff\x00\x00tj#\x00\x00\x00\x86\x02\x01\x00*`WU+\x00\x00\x00\x00\x00\x00\x00', 'n_type': 'NT_GNU_ABI_TAG', 'n_name': 'CORE', 'n_offset': 564, 'n_descsz': 144, 'n_size': 164, 'n_namesz': 5})
```

**Comment by eliben** [comment]
Could you please try to submit a pull request for this?

**Comment by zachriggle** [comment]
After my previous three pull requests were abandoned or rejected? No thanks.  However, this is still a valid bug report since the tag parsing is b0rken.

You can see how I fixed it here: https://github.com/binjitsu/binjitsu/blob/master/pwnlib/elf/corefile.py#L22-L48 and how I parsed the data here: https://github.com/binjitsu/binjitsu/blob/master/pwnlib/elf/corefile.py#L152-L176 

https://github.com/eliben/pyelftools/pull/64 <-- Rejected
https://github.com/eliben/pyelftools/pull/62 <-- Rejected
https://github.com/eliben/pyelftools/pull/46 <-- Not sure why I closed this.  Probably would have been rejected.

**Comment by eliben** [comment]
I'm sorry you feel this way about the other PRs. I think I provided objective review in them, so I don't know what I could have done differently.

**Comment by zachriggle** [comment]
I didn't mean to question your judgement, but just to state and explain that I've no intent to create a Pull Request, and have no fix to contribute to the project directly.  

I've hacked around it in a way which is sufficient for my usage in my project, and I've filed a bug to inform you that there is an error.

**Comment by eliben** [comment]
I believe this was addressed by #147; reopen if not
